clients/src/test/java/org/apache/kafka/clients/consumer/internals/AbstractStickyAssignorTest.java

```diff
@@ -1038,6 +1038,96 @@ public void testPartitionsTransferringOwnershipIncludeThePartitionClaimedByMulti
         assertTrue(isFullyBalanced(assignment));
     }
 
+    @ParameterizedTest(name = TEST_NAME_WITH_RACK_CONFIG)
+    @EnumSource(RackConfig.class)
+    public void testOwnedPartitionsAreStableForConsumerWithMultipleGeneration(RackConfig rackConfig) {
```

Comment:
nit: It seems that we also test the case where a partition is claimed by two consumers with different generation in this test. Should we try to reflect this in its name?

Reply:
I actually wonder if this was expected or if it is just a typo in the test case because there is also `testOwnedPartitionsAreInvalidatedForConsumerWithMultipleGeneration`.

Reply:
Ok I renamed the test to `testEnsurePartitionsAssignedToHighestGeneration` as the goal of this test is to make sure partitions are always assigned to the member with the highest generation.